magento/pwa-studio
# PWA-2295: Images to include dimensions

- Added shimmering effect for image content type
- Added test coverage for image shimmering effect



## Description

As a merchant I want to benefit from the freedom that Page Builder gives me in terms of authoring experience on the PWA storefront while having the best storefront performance so that page created with page builder loads with ~0 CLS in Google lighthouse

Acceptance Criteria

Images dimensions specific to the viewport
Image is loaded progressively into the space reserved according to this dimensions
Page created with content types that include images show increased performance and ~0 CLS

## Related Issue







Closes #PWA-2295

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






Verify pagebuilder content type images load progressively Example - Home page, Customer service, about-us
Verify Images load across venia app like Home, Category, Product, Search, Cart, Wishlist, Mini cart
Verify above on Chrome, Safari, FF and mobile device
Verify with different pwa-studio image configs - backend and onboard. (Also check fastly vs non-fastly env)
Run lighthouse and webpagetest
Verify existing cypress tests run with no issues

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
